AmericanMedicalAssociation/ama-style-guide-2
# EWL-8596: Prevents search forms from submitting if nothing is entered…

… into search field.



## Ticket(s)
**Do not** submit a Pull Request without a relevant JIRA ticket or Github issue! If you are creating a new pattern or feature, create an issue describing the need for this feature in Github **first** and then link to it below.

**Github Issue**
- N/A

**Jira Ticket**
- [EWL-8956: Prevent empty / "not set" searches from being run](https://issues.ama-assn.org/browse/EWL-8956)

## Description
Added script to prevent search form submission if no search term is entered or if a non-alphanumeric term is entered.


## To Test
- Link style guide with 'lando link-styleguide'
- In homepage search box, submit a search with nothing entered
- Confirm form does not submit and the placeholder text becomes bold

- enter a term with no letters or numbers **(NOTE: dashes were omitted from this exception considering terms such as covid-19)**
- Confirm form does not submit and the placeholder text becomes bold

- Enter a valid term, and confirm form does submit
- Repeat process for the search forms on the search page

## Visual Regressions

- n/a

## Relevant Screenshots/GIFs

![Screen Shot 2021-09-21 at 9 41 03 AM](https://user-images.githubusercontent.com/67962801/134192648-f1058bbb-ec70-454b-9c34-4b2b8ce7fb4c.png)


## Remaining Tasks
- n/a

## Additional Notes
- n/a

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)
